### PR TITLE
[Feature] remove EVM: fix unimplement in block_rs.[skip ci]

### PR DESCRIPTION
**Pull request by leeyr338**
- fix unimplement in block_rs.
- let Charge model unimplement.

### CODE REVIEW

File: cita-executor/core/src/libexecutor/block.rs [hunk 1]
```diff
@@ -50,14 +50,14 @@ lazy_static! {
...
-    fn drain(self) -> TrieDB<RocksDB>;
+    /// Drop this object.
+    fn drain(self);
 }
 
 pub struct ExecutedBlock {
     pub block: OpenBlock,
     pub receipts: Vec<Receipt>,
-    pub trie_db: Arc<TrieDB<RocksDB>>,
+    pub state: Arc<RefCell<CitaState<CitaTrieDB>>>,
```
Comment (kaikai1024) [inline cita-executor/core/src/libexecutor/block.rs]: Why use RefCell?

**Comment by kaikai1024** [comment]
Better to open a new pr to use one unified word: `trie_db`, `state`, or `state_db`.

File: cita-executor/core/src/libexecutor/block.rs [hunk 2]
```diff
@@ -50,14 +50,14 @@ lazy_static! {
 
 /// Trait for a object that has a state database.
 pub trait Drain {
-    /// Drop this object and return the underlieing database.
-    fn drain(self) -> TrieDB<RocksDB>;
+    /// Drop this object.
+    fn drain(self);
```
[the remark this answers is not in the document]
Comment (leeyr338) [reply cita-executor/core/src/libexecutor/block.rs]: I checked the place where called it, this function just clear a cache, it do not need to return a DB object now.

File: cita-executor/core/src/libexecutor/block.rs (hunk 1 above)
Comment (leeyr338) [reply cita-executor/core/src/libexecutor/block.rs]: cita-vm expected a RefCell, and I should build the cell at the beginning of opening an `executedBlock`. 
We should keep only one stateDB for the whole block transactions.

**Comment by leeyr338** [comment]
> Better to open a new pr to use one unified word: `trie_db`, `state`, or `state_db`.

Okay, I will do that later.

File: cita-executor/core/src/libexecutor/block.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-executor/core/src/libexecutor/block.rs]: > We should keep only one stateDB for the whole block transactions.

one struct? there're three trie: state, tx, receipt,

File: cita-executor/core/src/libexecutor/block.rs [hunk 3]
```diff
@@ -85,14 +85,20 @@ impl ExecutedBlock {
...
         conf: &BlockSysConfig,
         block: OpenBlock,
-        trie_db: Arc<TrieDB<RocksDB>>,
+        trie_db: Arc<CitaTrieDB>,
         state_root: H256,
         last_hashes: Arc<LastHashes>,
         eth_compatibility: bool,
     ) -> Result<Self, Error> {
+        let state = CitaState::from_existing(Arc::<CitaTrieDB>::clone(&trie_db), state_root)
+            .expect("Get state from trie db");
```
Comment (kaikai1024) [inline cita-executor/core/src/libexecutor/block.rs]: Maybe `expect` is not a good idea. It'll panic. That's not what we want

Comment (leeyr338) [reply cita-executor/core/src/libexecutor/block.rs]: But, if it cannot get state correctly, the logic can do nothing!

**Pull request merged**